bcgov/HOUS-permit-portal
# Fix issues re-inviting users

## Description

Fixes issues inviting a user that already has a submitter account.

Previously, if a review staff member was invited but did not accept the
invite and instead created a submitter account, they could not always be
re-invited as a review staff. This occurred if the submitter was
retrieved first when inviting by email. In this scenario, the submitter
would be updated to have the same nickname as the existing review staff.
This triggered an error since the nickname field has a unique
constraint. This exposed the real error, which is that the submitter
should not be considered when retrieving the user to invite.

This also fixes a related issue where a submitter is invited as a review
staff for the first time. Instead of sending an invite to the review
staff as a new user, the submitter was automatically converted to a
review staff without needing to accept the invite.

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

[BPHH-1636](https://hous-bssb.atlassian.net/browse/BPHH-1636)

## Steps to QA

1. Login as a super admin or review manager and invite a reviewer or review manager. Note the email.
2. In a separate session, create a submitter account. Confirm the profile, setting the notification email to the email that was used in the invitation in step 1.
3. As the super admin or review manager, try to re-invite the invited user from step 1. The invitation should send.
4. Open the invitation in a separate session and sign in with the same BCeID used in step 2. 

After accepting the invitation, the user should be prompted to accept the EULA and then be logged in as the invited user type (reviewer or review manager). 

## UI Accessibility Checklist



_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?

## General Checklist


## [optional] Are there any post-deployment tasks we need to perform?

N/A
